<?xml version="1.0" ?>

<launch>
    <arg name="robot_type" default="wheel_legged"/>
    <arg name="rviz" default="true"/>
    <arg name="cheater" default="false"/>
    <arg name="multiplot"          default="false"/>

      <!-- <include file="$(find wheel_legged_controllers)/launch/joy_teleop.launch"/> -->

    <!-- make the files into global parameters -->
    <param name="urdfFile" value="$(find wheel_legged_description)/urdf/Wheel_legged_sim2.urdf"/>
    <param name="taskFile" value="$(find wheel_legged_controllers)/config/Wheel_legged_command/task.info"/>
    <param name="referenceFile" value="$(find wheel_legged_controllers)/config/Wheel_legged_command/reference.info"/>
    <param name="gaitCommandFile" value="$(find wheel_legged_controllers)/config/Wheel_legged_command/gait.info"/>
    <param name="wheel_legged_description" textfile="$(find wheel_legged_description)/urdf/Wheel_legged_sim2.urdf" />

    <rosparam file="$(find wheel_legged_controllers)/config/controllers.yaml" command="load"/>

    <!-- <node if="$(arg rviz)" name="rviz" pkg="rviz" type="rviz" args="-d $(find wheel_legged_description)/rviz/test_rviz_v2.0.rviz" required="true" /> -->


    <node pkg="wheel_legged_dummy" type="wheel_legged_robot_gait_command" name="Wheel_legged_robot_gait_command"
          output="screen" launch-prefix="gnome-terminal --"/>

    <node pkg="wheel_legged_controllers" type="wheel_legged_target_trajectories_publisher" name="Wheel_legged_robot_target"
          output="screen"/>

    <node pkg="wheel_legged_controllers" type="normal_controller_node" name="normal_controller"
          output="screen" launch-prefix="gnome-terminal --"/>

    <node pkg="mujoco_sim" type="wheel_legged_sim.py" name="wheel_legged_sim"
          output="screen"/>

      
</launch>